### PR TITLE
vokoscreen-beta: init at 2.5.8

### PR DESCRIPTION
**Pull request by GRBurst**
###### Motivation for this change
Only the stable version of vokoscreen is available, but is missing multiple features that I needed.
I already build and installed it locally and used it multiple times successfully :-)
I guess it could be bundled with the stable version, but this may be more reasonable when the beta and stable version will be maintained continuously.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jtojnar** [comment]
The stable version is ancient, I would just update the existing package.

**Review by aanderse — CHANGES_REQUESTED** [review]
Just a couple nitpicks I noticed

File: pkgs/applications/video/vokoscreen-beta/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub
+, pkgconfig, qtbase, qttools, qmake, qtmultimedia, qtx11extras, alsaLib, libv4l, libXrandr
+, ffmpeg
+}:
+
+stdenv.mkDerivation rec {
+
+  version = "2.5.8-beta";
```
[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/applications/video/vokoscreen-beta/default.nix]: It is a release: https://github.com/vkohaupt/vokoscreen/releases/tag/2.5.8-beta

[the remark this answers is not in the document]
Comment (aanderse) [reply pkgs/applications/video/vokoscreen-beta/default.nix]: My bad

**Comment by GRBurst** [comment]
@jtojnar I am not sure whether this makes sense to replace a `stable` version with a `beta`. I even made sure to replace the target binary to be "vokoscreen-beta" in order to enable having both versions, `stable` and `beta`, installed. I agree that the stable is ancient and I can tell that I successfully used the beta a few times now, but from a conceptional point of view I would still disagree :sweat_smile:

File: pkgs/applications/video/vokoscreen-beta/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    sed -i 's/TARGET\ =\ vokoscreen/TARGET\ =\ vokoscreen-beta/g' vokoscreen.pro
+  '';
+
+  postConfigure = ''
+    substituteInPlace settings/QvkSettings.cpp --subst-var-by ffmpeg ${ffmpeg}
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Simple GUI screencast recorder, using ffmpeg";
+    homepage = http://linuxecke.volkoh.de/vokoscreen/vokoscreen.html;
```
[the remark this answers is not in the document]
Comment (GRBurst) [reply pkgs/applications/video/vokoscreen-beta/default.nix]: yes, u r right

File: pkgs/applications/video/vokoscreen-beta/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchFromGitHub
+, pkgconfig, qtbase, qttools, qmake, qtmultimedia, qtx11extras, alsaLib, libv4l, libXrandr
+, ffmpeg
+}:
+
+stdenv.mkDerivation rec {
+
+  version = "2.5.8-beta";
+  name = "vokoscreen-${version}";
```
[the remark this answers is not in the document]
Comment (GRBurst) [reply pkgs/applications/video/vokoscreen-beta/default.nix]: sure thing

**Comment by jtojnar** [comment]
A beta package increases the maintenance costs, especially keeping the two expressions in sync.

Is there any reason why someone would need the old version (e.g. removed functionality)? In practice, we often see that while a release was not yet labelled stable, it is actually more stable than the most recent stable build, thanks to newly added bug fixes. For that reason, many packages in nixpkgs often use “unstable” releases or even VCS snapshots.

But really, it needs to be judged on case by case basis. If the beta introduces more bugs than it fixes then parallel expression would be justified. In that case both expressions should share a directory and perhaps inherit from single common expression.

**Comment by GRBurst** [comment]
@jtojnar well okay, I will make a new pr that upgrades the stable to the beta :wink:

**Pull request closed without merging**